core/src/main/java/amino/run/compiler/PolicyStub.java

```diff
@@ -296,15 +297,22 @@ public String getMethodContent(MethodStub m, boolean isDMMethod) {
      * Returns the KernelRPC stub implementation code based on whether the method is a DM method or
      * Application method.
      *
+     * @param m
      * @param isDMMethod
      * @return Stub code for RPC call based on method type.
      */
-    public String getMethodRPCContent(boolean isDMMethod) {
-        if (isDMMethod == true) {
+    public String getMethodRPCContent(MethodStub m, boolean isDMMethod) {
+        if (m.retType.equals(Void.TYPE)) {
```

Comment:
It is better practice to reverse this condition like this - Void.TYPE.equals(m.retType) so as to avoid the NPE. I know that it is not null at least for this case. See the comment below too.